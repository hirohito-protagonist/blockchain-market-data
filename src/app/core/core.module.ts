import { NgModule, ModuleWithProviders } from '@angular/core';
import { StoreModule } from '@ngrx/store';

import { CoreAppContainerComponent } from './containers/app.container';

export {
  CoreAppContainerComponent
};

import { reducers } from './reducers/index.reducer';

@NgModule({
  imports: [
    StoreModule.forFeature('core', reducers)
  ],
  declarations: [ CoreAppContainerComponent ]
})
export class CoreModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: CoreModule,
      providers: []
    };
  }
}
