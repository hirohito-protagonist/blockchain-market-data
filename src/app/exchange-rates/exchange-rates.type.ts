export interface ConvertToBTC {
  value: string|number;
  currency: string;
}

interface ConvertToBtcData extends ConvertToBTC {
  btc: number;
}

export interface ConvertBtcState {
  isFetching: boolean;
  convert: ConvertToBtcData;
  lastUpdate: number;
}

export interface MarketPrices {
  [key: string]: {
    '15m': number;
    'last': number;
    'buy': number;
    'sell': number;
    'symbol': string;
  };
}
