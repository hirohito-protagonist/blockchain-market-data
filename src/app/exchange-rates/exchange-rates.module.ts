import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { reducers } from './reducers/index.reducer';

import { ExchangeRatesContainerComponent } from './containers/exchange-rates.container';
import { MarketPricesViewComponent } from './views/market-prices.view';
import { ConverterViewComponent } from './views/converter-view';

import { MarketPricesEffects } from './effects/market-prices.effect';
import { ConvertBtcEffects } from './effects/convert-btc.effect';


@NgModule({
  imports: [
    StoreModule.forFeature('exchangeRates', reducers),
    EffectsModule.forFeature([MarketPricesEffects, ConvertBtcEffects])
  ]
})
export class RootExchangeRateModule {}


@NgModule({
  imports: [CommonModule],
  declarations: [
    ExchangeRatesContainerComponent,
    MarketPricesViewComponent,
    ConverterViewComponent
  ],
  exports: [ ExchangeRatesContainerComponent ]
})
export class ExchangeRatesModule {

  static forRoot(): ModuleWithProviders {

    return {
      ngModule: RootExchangeRateModule,
      providers: []
    };
  }
}
