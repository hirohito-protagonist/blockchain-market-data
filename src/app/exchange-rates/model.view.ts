import { ConvertBtcState } from './exchange-rates.type';

export interface ExchangeRatesViewModel {
  update: number;
  data: {
    marketPrices: any;
    currencies: {
      currency: string;
      symbol: string;
    }[];
    convert: ConvertBtcState;
  };
}
