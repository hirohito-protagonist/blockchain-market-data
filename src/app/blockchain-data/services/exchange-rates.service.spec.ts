import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { TestBed } from '@angular/core/testing';

import { ExchangeRatesService } from './exchange-rates.service';
import { fromBlockchainDataMarketPriceEntity } from './../entities/index.entity';

describe('ExchangeRatesService', () => {

  let service: ExchangeRatesService;
  let httpTestingController: HttpTestingController;

  beforeEach(() => {

    TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule
      ],
      providers: [
        ExchangeRatesService
      ]
    });

    service = TestBed.get(ExchangeRatesService);
    httpTestingController = TestBed.get(HttpTestingController);
  });

  afterEach(() => {

    httpTestingController.verify();
  });

  describe('ticker', () => {

    it('should return market prices', () => {

      const testData: fromBlockchainDataMarketPriceEntity.MarketPricesEntities = {
        'USD' : { '15m' : 478.68, 'last' : 478.68, 'buy' : 478.55, 'sell' : 478.68,  'symbol' : '$' }
      };

      service.ticker().subscribe((responseBody) => {

        expect(responseBody).toMatchSnapshot();
      });

      const req = httpTestingController.expectOne('https://blockchain.info/ticker?cors=true');

      expect(req.request.method).toMatchSnapshot();
      req.flush(testData);
    });
  });

  describe('tobtc', () => {

    it('should convert x value in the provided currency to btc', () => {


      service.tobtc('USD', 500).subscribe(btc => {

        expect(btc).toMatchSnapshot();
      });

      const req = httpTestingController.expectOne('https://blockchain.info/tobtc?cors=true&currency=USD&value=500');

      expect(req.request.method).toMatchSnapshot();
      req.flush(10);
    });
  });
});
