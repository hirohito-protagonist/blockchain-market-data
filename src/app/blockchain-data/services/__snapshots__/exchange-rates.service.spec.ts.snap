// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRatesService ticker should return market prices 1`] = `"GET"`;

exports[`ExchangeRatesService ticker should return market prices 2`] = `
Object {
  "USD": Object {
    "15m": 478.68,
    "buy": 478.55,
    "last": 478.68,
    "sell": 478.68,
    "symbol": "$",
  },
}
`;

exports[`ExchangeRatesService tobtc should convert x value in the provided currency to btc 1`] = `"GET"`;

exports[`ExchangeRatesService tobtc should convert x value in the provided currency to btc 2`] = `10`;
